accepted/2022/version-selection.md

```diff
@@ -0,0 +1,449 @@
+
+## Intuitive version selection
+
+.NET version selection includes cases of being both overly strict and overly lax. Version selection should have a baseline intutitive behavior and provide (as appropriate) additional experiences to meet niche needs.
+
+The most problematic behaviors today are:
+
+- [Multi-level lookup](https://github.com/dotnet/sdk/issues/12353) can result in the global install location being used when a private location is intended.
+- `global.json` defaults to no roll-forward by default, for the specified SDK version. This results in very fragile environments, by default. Instead, the version specified should be a version floor by default.
+
+## Convenient and consistent infra builds
+
+As a (.NET) team, we get significant feedback about CI services, primarily Azure DevOps and GitHub Actions. Customers rely on these services very heavily for their builds, for projects big and small. A key aspect of CI is automatic and managed delivery of developer platforms (like .NET). Developers expect same-day delivery of .NET in CI. If there is a public blog post about a new .NET version, that new version should be available in CI with very low or (ideally) no effort on the part of developers.
+
+As we were preparing to ship .NET 6, we took a look at the [GitHub Actions Virtual Environments VM images](https://github.com/actions/virtual-environments). We saw that the [Actions Ubuntu 20.04 images](https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu2004-Readme.md), for example, included [several versions of the .NET SDK](https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu2004-Readme.md#net-core-sdk), some of which are out-of-support. That caused us to re-assess our approach to these CI images.
+
+At the time of writing (February 2022), the following .NET SDK versions are available in those images:
+
+- `2.1.302`, `2.1.403`, `2.1.526`, `2.1.617`, `2.1.701`, `2.1.818`
+- `3.1.120`, `3.1.202`, `3.1.302`, `3.1.416`
+- `5.0.104`, `5.0.210`, `5.0.303`, `5.0.404`
+- `6.0.101`
+
+The .NET SDK has a concept of ["version compatibility bands"](https://docs.microsoft.com/dotnet/core/porting/versioning-sdk-msbuild-vs). Those are all the hundreds-based versions, like `2.1.3xx` and `2.1.4xx`. By default, if you specify `2.1.300` (or `2.1.317`), your builds requires a `2.1.3xx` SDK (at least as high as the version number specified), but will not work with `2.1.400` or higher versions. This behavior is what motivates (one could argue, requires) the GitHub team to maintain all these .NET SDK versions so that customer builds work, even out-of-support ones. It is easy to come to the conclusion that this is a problematic design, leading to unfortunate requirements on the part of service providers like GitHub.
```

Comment:
We call these feature bands in other documents. You might want to use the same language here rather than compatibility bands.

Reply:
Yup. Fixed.